src/commands/export-sql.ts

```diff
@@ -365,20 +416,20 @@ export class ExportSQLCommand {
 
 		const { latestBackup } = await fetchLatestBackupAndJobStatus( this.app.id, this.env.id );
 
+		if ( ! latestBackup ) {
+			await this.track( 'error', {
+				error_type: 'no_backup_found',
+				error_message: 'No backup found for the site',
+			} );
+			exit.withError( `No backup found for site ${ this.app.name }` );
+		}
+
 		if ( ! this.generateBackup ) {
-			if ( ! latestBackup ) {
-				await this.track( 'error', {
-					error_type: 'no_backup_found',
-					error_message: 'No backup found for the site',
-				} );
-				exit.withError( `No backup found for site ${ this.app.name }` );
-			} else {
-				console.log(
-					`${ getGlyphForStatus( 'success' ) } Latest backup found with timestamp ${
-						latestBackup.createdAt
-					}`
-				);
-			}
+			console.log(
+				`${ getGlyphForStatus( 'success' ) } Latest backup found with timestamp ${
+					latestBackup.createdAt
+				}`
+			);
```

Comment:
This is probably the biggest logic change. I did this so that `latestBackup` is inferred to not be nullish.